### ISSUE TITLE
Refactor: Phase transition handling cleanup & auto-batch resizing

### ISSUE THREAD

**Issue by Lev-Stambler** [DeepOpt-com/WrinkleFree-Monorepo#30]
## Summary

Phase/stage transitions in training are handled across multiple components with inconsistent patterns, incomplete features, and leftover debug code. This issue tracks a general cleanup of phase transition handling, including auto-batch resizing.

## Current Architecture

```
┌─────────────────────────────────────────────────────────────┐
│              Phase Transition Components                    │
├─────────────────────────────────────────────────────────────┤
│  CurriculumScheduler     │ Weight interpolation between     │
│  (manager.py:58-174)     │ phases (step/linear/cosine)      │
├──────────────────────────┼──────────────────────────────────┤
│  ObjectiveManager        │ Applies weights, conditional     │
│  (manager.py:176-472)    │ DLM masking, curriculum stepping │
├──────────────────────────┼──────────────────────────────────┤
│  Lazy Teacher Loading    │ Defers teacher until distill     │
│  (module.py:148-310)     │ weight > 0, reduces batch size   │
├──────────────────────────┼──────────────────────────────────┤
│  LambdaWarmupCallback    │ Gradual quantization warmup      │
│  (callbacks.py:400-463)  │ (parallel to curriculum)         │
├──────────────────────────┼──────────────────────────────────┤
│  MetaOptimizerCallback   │ LDC-MTL/ODM/LayerLR overlays     │
│  (meta/callback.py)      │ can override curriculum weights  │
└─────────────────────────────────────────────────────────────┘
```

### 1. Auto-Batch Resizing Only at Startup
**Files:** `train_lightning.py:864-886`, `module.py:211-248`

- `auto_batch_size=true` runs Tuner once at startup
- Teacher lazy loading has its own `_reduce_batch_size()` (hardcoded 0.5 factor)
- No re-tuning when objectives activate or memory requirements change
- `_reduce_batch_size()` calls `trainer.reset_train_dataloader()` - untested with FSDP

### 2. Debug Logging Left in Production Code
**Files:** `module.py:274, 431-437, 448, 457`

```python
# Examples of debug prints that should be removed:
print(f"[DEBUG] Optimizer config: ...")
print(f"[DEBUG] Final learning rate: ...")
```

### 3. Incomplete Data Config Switching
**Files:** `manager.py:36, 163-165`

- `CurriculumPhase.data_config` field exists but is never used
- Intent was to switch datasets between phases, but feature is incomplete
- Either implement or remove the dead field

### 4. Curriculum + Lambda Warmup Uncoordinated
**Files:** `train_lightning.py:578-585`, `callbacks.py:400-463`

- Both can be active simultaneously with no coordination
- Both ramp up in parallel - unclear if intentional
- Should document expected interaction or add coordination

### 5. DLM + Logits Distill Incompatibility Warning Only
**Files:** `manager.py:220-236`

- DLM uses masked prediction (no shift), logits distill uses AR shift
- Config allows this incompatible combination - only warns at runtime
- Should validate in config or factory, not just warn

### 6. Teacher Config Validation Too Late
**Files:** `train_lightning.py:398-438`, `module.py:207-209`

- No validation that teacher model is accessible until lazy load
- Silent failure: returns False, training continues without distillation
- Should validate early or fail loudly

### 7. Checkpoint Resume Edge Cases
**Files:** `module.py:872-904`

- `curriculum_state` may not exist in early checkpoints
- Falls back gracefully but could be more robust
- `_last_curriculum_step` tracking added but not fully tested

### 8. Deferred Initialization Complexity
**Files:** `meta/callback.py:147-162`, `callbacks.py:23-72`

- LayerLR init deferred to `on_train_start()` for BatchSizeFinder compatibility
- MuonClipInitCallback re-registers hooks after BatchSizeFinder cycles eval/train
- Complex interaction patterns that are fragile

### Phase 1: Cleanup
- [ ] Remove debug print statements (`module.py`)
- [ ] Remove or implement `data_config` field (`manager.py:36`)
- [ ] Add config-level validation for DLM + logits_distill (`factory.py`)
- [ ] Validate teacher model accessibility in `prepare_teacher_config()`

### Phase 2: Auto-Batch Resizing
- [ ] Extract batch size adjustment to reusable utility
- [ ] Hook into `ObjectiveManager.step_curriculum()` to detect phase changes
- [ ] Re-run batch size tuning when memory-intensive objectives activate
- [ ] Add `batch_size_factor` per curriculum phase (optional override)
- [ ] Test with FSDP or add explicit single-GPU guard

### Phase 3: Documentation & Coordination
- [ ] Document curriculum + lambda warmup interaction
- [ ] Add architecture diagram to `docs/ai-code/reference.md`
- [ ] Consider unified "PhaseTransitionManager" abstraction

## Key File References

| Component | File | Lines | Issue |
|-----------|------|-------|-------|
| Curriculum stepping | `module.py` | 368-376 | Works correctly |
| Lazy teacher load | `module.py` | 148-209 | Silent failure |
| Batch reduction | `module.py` | 211-248 | Untested FSDP |
| Debug prints | `module.py` | 274, 431-437 | Remove |
| Data config field | `manager.py` | 36, 163-165 | Dead code |
| DLM validation | `manager.py` | 220-236 | Warn → Error |
| Curriculum factory | `factory.py` | 127-162 | Add validation |
| Lambda warmup | `callbacks.py` | 400-463 | Document |
| Meta deferred init | `meta/callback.py` | 147-162 | Fragile |

## Related Commits
- `e5af817` - Lazy load teacher model when distill weight becomes non-zero
- `baad86a` - Stabilize MuonClip (re-registration after BatchSizeFinder)
